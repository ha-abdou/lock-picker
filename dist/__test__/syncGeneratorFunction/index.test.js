"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const __1 = __importDefault(require(".."));
function* names(name) {
    yield `lolo`;
    yield `toto`;
    yield `fofo`;
}
function* lang() {
    yield "fr";
    yield "en";
    yield "it";
}
function* tags() {
    yield "tag1";
    yield "tag2";
}
function* tag(tag) {
    yield (name) => `${name} ${tag}`;
    yield (name) => `${name} ${tag}`;
}
describe("Sync generator functions", () => {
    test("simple", () => {
        const exp = `$lang() + $lang()`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addGeneratorFunction("$lang", lang);
        })).toMatchSnapshot();
    });
    test("simple 2", () => {
        const exp = `$lang() + "/" + $names() + "/" + $lang()`;
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addGeneratorFunction("$lang", lang);
            lockPicker.addGeneratorFunction("$names", names);
        })).toMatchSnapshot();
    });
    test("nested", () => {
        const exp = '$lang() + "/" + tag(tags())(names())';
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addGeneratorFunction("$lang", lang);
            lockPicker.addGeneratorFunction("names", names);
            lockPicker.addGeneratorFunction("tags", tags);
            lockPicker.addGeneratorFunction("tag", tag);
        })).toMatchSnapshot();
    });
    test("nested 2", () => {
        const exp = '$lang() + "/" + tag($lang())(names())';
        expect(__1.default(exp, (lockPicker) => {
            lockPicker.addGeneratorFunction("$lang", lang);
            lockPicker.addGeneratorFunction("names", names);
            lockPicker.addGeneratorFunction("tag", tag);
        })).toMatchSnapshot();
    });
});
//# sourceMappingURL=index.test.js.map