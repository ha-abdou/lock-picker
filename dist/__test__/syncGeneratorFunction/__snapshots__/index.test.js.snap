// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sync generator functions nested 1`] = `
Array [
  "fr/lolo tag1",
  "en/lolo tag1",
  "it/lolo tag1",
  "fr/lolo tag1",
  "en/lolo tag1",
  "it/lolo tag1",
  "fr/lolo tag2",
  "en/lolo tag2",
  "it/lolo tag2",
  "fr/lolo tag2",
  "en/lolo tag2",
  "it/lolo tag2",
  "fr/toto tag1",
  "en/toto tag1",
  "it/toto tag1",
  "fr/toto tag1",
  "en/toto tag1",
  "it/toto tag1",
  "fr/toto tag2",
  "en/toto tag2",
  "it/toto tag2",
  "fr/toto tag2",
  "en/toto tag2",
  "it/toto tag2",
  "fr/fofo tag1",
  "en/fofo tag1",
  "it/fofo tag1",
  "fr/fofo tag1",
  "en/fofo tag1",
  "it/fofo tag1",
  "fr/fofo tag2",
  "en/fofo tag2",
  "it/fofo tag2",
  "fr/fofo tag2",
  "en/fofo tag2",
  "it/fofo tag2",
]
`;

exports[`Sync generator functions nested 2 1`] = `
Array [
  "fr/lolo fr",
  "fr/lolo fr",
  "en/lolo en",
  "en/lolo en",
  "it/lolo it",
  "it/lolo it",
  "fr/toto fr",
  "fr/toto fr",
  "en/toto en",
  "en/toto en",
  "it/toto it",
  "it/toto it",
  "fr/fofo fr",
  "fr/fofo fr",
  "en/fofo en",
  "en/fofo en",
  "it/fofo it",
  "it/fofo it",
]
`;

exports[`Sync generator functions simple 1`] = `
Array [
  "frfr",
  "enen",
  "itit",
]
`;

exports[`Sync generator functions simple 2 1`] = `
Array [
  "fr/lolo/fr",
  "fr/toto/fr",
  "fr/fofo/fr",
  "en/lolo/en",
  "en/toto/en",
  "en/fofo/en",
  "it/lolo/it",
  "it/toto/it",
  "it/fofo/it",
]
`;
